src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,52 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private Node<T> top;
+    private int size;
 
+    public void push(T value) {
+        Node<T> node;
+        if (size == 0) {
+            node = new Node<>(value, null);
+        } else {
+            node = new Node<>(value, top);
+        }
+        top = node;
+        size++;
     }
 
-    public T peek() {
-        return null;
+    public T peek() throws EmptyStackException {
```

Comment:
```suggestion
    public T peek() {
```
is it checked or unchecked?

Reply:
it's unchecked

Reply:
than we don't need to even process it with try-catch, not to mention throwing it up